iree/compiler/Dialect/Util/IR/UtilOps.td

```diff
@@ -176,6 +176,46 @@ def Util_RangeExtentsOp : Util_PureOp<"range.extents", [
   let hasCanonicalizer = 1;
 }
 
+//===----------------------------------------------------------------------===//
+// Alignment Arithmetic
+//===----------------------------------------------------------------------===//
+
+def Util_AlignOp : Util_PureOp<"align", [
+  SameOperandsAndResultType
+  ]> {
+  let summary = "Aligns up to a power-of-two alignment if required";
+  let description = [{
+     Aligns |value| up to the given power-of-two |alignment| if required.
+  }];
+
+  let arguments = (ins
+    SignlessIntegerLike:$value,
+    SignlessIntegerLike:$alignment
+  );
+
+  let results = (outs
+    SignlessIntegerLike:$result
+  );
+
+  let assemblyFormat = [{
+    $value `,` $alignment attr-dict `:` type($result)
+  }];
+
+  let builders = [
+    OpBuilder<(ins
+      "Value":$value,
+      "int64_t":$alignment
+    ),
+    [{
+      build($_builder, $_state, value.getType(), value,
+       $_builder.createOrFold<arith::ConstantIndexOp>($_state.location, alignment));
+    }]>,
+  ];
+
+  //TODO(#5405) for adding canonicalizers/folders
```

Comment:
use standard formatting for this: `// TODO(#5405): add canonicalizers/folders to do XXX.`
(the spaces and colon are important)

Specifying some examples of the kind of canonicalizers/folders is also useful to readers coming across this. For example, if either the alignment or the value are constant, or known to be within a certain range (not zero, etc) then it may simplify.

Another important one (and one of the motivators for having this op) that could be done without too much work is removing duplicate alignments - util.align 16 -> util.align 4 should just fold to util.align 16, for example.